src/test/kotlin/no/nav/syfo/dialogmote/api/v2/TildelDialogmoteApiV2Spek.kt

```diff
@@ -0,0 +1,239 @@
+                externalMockEnvironment.environment.aadAppClient,
+                externalMockEnvironment.wellKnownVeilederV2.issuer,
+                veilederCallerIdent,
+            )
+
+            application.testApiModule(
+                externalMockEnvironment = externalMockEnvironment,
+                altinnMock = altinnMock,
+                esyfovarselProducer = esyfovarselProducerMock,
+            )
+
+            beforeEachGroup {
+                database.dropData()
+            }
+            beforeEachTest {
+                val altinnResponse = ReceiptExternal()
+                altinnResponse.receiptStatusCode = ReceiptStatusEnum.OK
+
+                justRun { esyfovarselProducerMock.sendVarselToEsyfovarsel(esyfovarselHendelse) }
+                clearMocks(altinnMock)
+                every {
+                    altinnMock.insertCorrespondenceBasicV2(any(), any(), any(), any(), any())
+                } returns altinnResponse
+            }
+            afterGroup { database.dropData() }
+
+            describe("Tildel dialogmoter") {
+                val urlMote = "$dialogmoteApiV2Basepath$dialogmoteApiPersonIdentUrlPath"
+                val urlMoterEnhet = "$dialogmoteApiV2Basepath$dialogmoteApiEnhetUrlPath/${UserConstants.ENHET_NR.value}"
+                val urlTildelMote = "$dialogmoteApiV2Basepath$dialogmoteTildelPath"
+                val newDialogmote = generateNewDialogmote(UserConstants.ARBEIDSTAKER_FNR)
+                val newDialogmoteAnnenArbeidstaker = generateNewDialogmote(UserConstants.ARBEIDSTAKER_ANNEN_FNR)
+
+                describe("Happy path") {
+                    it("should tildele dialogmoter if request is successful") {
+                        val createdDialogmoterUuids = mutableListOf<UUID>()
+                        database.connection.run { this.createNewDialogmoteWithReferences(newDialogmote) }
+                        database.connection.run { this.createNewDialogmoteWithReferences(newDialogmoteAnnenArbeidstaker) }
+
+                        with(
+                            handleRequest(HttpMethod.Get, urlMoterEnhet) {
+                                addHeader(HttpHeaders.Authorization, bearerHeader(veilederCallerToken))
+                            }
+                        ) {
+                            response.status() shouldBeEqualTo HttpStatusCode.OK
+                            val dialogmoteList = objectMapper.readValue<List<DialogmoteDTO>>(response.content!!)
+
+                            dialogmoteList.size shouldBeEqualTo 2
+                            dialogmoteList.any { dialogmoteDTO -> dialogmoteDTO.tildeltVeilederIdent == veilederCallerIdent } shouldBeEqualTo true
+                            dialogmoteList.any { dialogmoteDTO -> dialogmoteDTO.tildeltVeilederIdent == veilederIdentTildelesMoter } shouldBeEqualTo false
+
+                            createdDialogmoterUuids.addAll(dialogmoteList.map { UUID.fromString(it.uuid) })
+                        }
```

Comment:
Er denne her bare for å sjekke at det funket å legge inn i databasen?

Reply:
Yup, har kopiert en del over fra `OvertaDialogmoteApiV2Spek.kt` så kan godt fjerne det. Virket logisk å sjekke status før og etter tildel post kallet.

Reply:
Akkurat her ble det litt stress å bare hente fra db. Funksjonen for å gjøre om fra P-objekt til domene objekt var ganske omfattende 😿
Vi kommer til å få muligheten for å skrive disse testene bedre når vi gjør om til Repository klasser tror jeg 😄